### PR TITLE
(WIP) Update README with steps to set up PortaFly environment

### PR DESCRIPTION
**Pull request by cntlsn**
**What this PR does / why we need it**:

Add steps to set up the PortaFly environment

**Which issue(s) this PR fixes** 

https://issues.redhat.com/browse/THREESCALE-5629

**Notes for the reviewers**
Please follow the steps in the README to set up the environment and run PortaFly on your machine.

### CODE REVIEW

File: portafly/README.md [hunk 1]
```diff
@@ -1,6 +1,31 @@
 # PortaFly
 A React SPA crafted with the ambitious purpose of replacing Porta UI, currently residing inside the Rails App.
 
+## Getting up and running
+1. Clone the `porta` repository to your local machine
+2. Create a `.env` file:
```
Comment (damianpm) [inline portafly/README.md:6]: Creating an `.env` file is optional, you could create the file or simply add the Environment variables in the terminal, before `yarn start`, look [here](https://github.com/3scale/porta/blob/master/portafly/README.md#environment-variables)

File: portafly/README.md [hunk 2]
```diff
@@ -1,6 +1,31 @@
...
+  * Create a new file in the `/portafly` root folder, and name it `.env`
+  * Paste `REACT_APP_API_HOST="https://multitenant-admin.preview01.3scale.net"` into it, and Save
```
Comment (damianpm) [inline portafly/README.md:8]: This is only needed of you are creating your access token in the preview environment, you could create it in your local environment

File: portafly/README.md [hunk 3]
```diff
@@ -1,6 +1,31 @@
...
+3. Create an Access token:
```
Comment (damianpm) [inline portafly/README.md:9]: This step is right (by now), but it's not the definitive way to authenticate, just a temporary solution. IDK if we should add this section, as this is going to change in the future...

File: portafly/README.md [hunk 4]
```diff
@@ -1,6 +1,31 @@
...
+  * Access your local Porta build (or access [Preview](https://multitenant-admin.preview01.3scale.net/))
+  * Navigate to Settings > Personal > Tokens
+  * Click the "Add Access Token" button
+  * Give your access token a Name
+  * Select all scopes
+  * Set permissions to Read & Write
+  * Create the access token
+  * Copy the access token
+4. Allow CORS requests in your browser:
```
Comment (damianpm) [inline portafly/README.md:18]: We may explain that allowing CORS is only needed in development mode, and add a disclaimer that if you decide to add an extension to bypass CORS, is at your own risk

File: portafly/README.md [hunk 5]
```diff
@@ -1,6 +1,31 @@
...
+  * Install the CORS browser extension ([Google Chrome extension](https://chrome.google.com/webstore/detail/allow-cors-access-control/lhobafahddgcelffkeicbaginigeejlf))
+  * Click on the extension icon in the browser  toolbar and activate it by clicking on its logo
+5. Run PortaFly: 
+  * Open a terminal session
+  * Navigate to `/portafly`
+  * Run `yarn install`
```
Comment (damianpm) [inline portafly/README.md:24]: This is right, but is repeated [here](https://github.com/3scale/porta/blob/master/portafly/README.md#yarn-start)

File: portafly/README.md (hunk 2 above)
Comment (Martouta) [inline portafly/README.md:8]: This is very unstable. We use Preview for testing and many times this API host won't be available. I think that it is important to point this out.

Comment (Martouta) [reply portafly/README.md:8]: Btw, it should be possible to use the local porta master/tenant account as API host as well. In fact, that will be more stable.

File: portafly/README.md [hunk 6]
```diff
@@ -1,6 +1,31 @@
 # PortaFly
 A React SPA crafted with the ambitious purpose of replacing Porta UI, currently residing inside the Rails App.
 
+## Getting up and running
+1. Clone the `porta` repository to your local machine
+2. Create a `.env` file:
+  * Create a new file in the `/portafly` root folder, and name it `.env`
+  * Paste `REACT_APP_API_HOST="https://multitenant-admin.preview01.3scale.net"` into it, and Save
+3. Create an Access token:
+  * Access your local Porta build (or access [Preview](https://multitenant-admin.preview01.3scale.net/))
```
Comment (Martouta) [inline portafly/README.md:10]: This should be in the same host used in the previous step.

File: portafly/README.md [hunk 7]
```diff
@@ -1,6 +1,31 @@
...
+  * Navigate to Settings > Personal > Tokens
+  * Click the "Add Access Token" button
+  * Give your access token a Name
+  * Select all scopes
+  * Set permissions to Read & Write
+  * Create the access token
+  * Copy the access token
+4. Allow CORS requests in your browser:
+  * Install the CORS browser extension ([Google Chrome extension](https://chrome.google.com/webstore/detail/allow-cors-access-control/lhobafahddgcelffkeicbaginigeejlf))
```
Comment (Martouta) [inline portafly/README.md:19]: It is ok to reference to the chrome extension, but imo, we should also reference to [the firefox one](https://addons.mozilla.org/en-US/firefox/addon/cors-everywhere/).
specially now that we are considering running our cukes with firefox: https://github.com/3scale/porta/pull/2039

File: portafly/README.md (hunk 1 above)
Comment (didierofrivia) [reply portafly/README.md:6]: yeah, there are options like adding it to your .zshrc / .bashrc, explicitely declare them before executing the command, etc

File: portafly/README.md (hunk 2 above)
Comment (didierofrivia) [reply portafly/README.md:8]: true, maybe a way to describe the necessity of having an instance running of Porta project. Also, our preview env shouldn't be public.

Comment (didierofrivia) [reply portafly/README.md:8]: we shouldn't make our preview env public

File: portafly/README.md [hunk 8]
```diff
@@ -1,6 +1,31 @@
...
+  * Click on the extension icon in the browser  toolbar and activate it by clicking on its logo
+5. Run PortaFly: 
+  * Open a terminal session
+  * Navigate to `/portafly`
+  * Run `yarn install`
+  * Run `yarn start`
+  * Open [http://localhost:3003](http://localhost:3003) (if it hasn't opened automagically)
```
Comment (porueesq) [inline portafly/README.md:26]:
```suggestion
  * If the home page hasn't been displayed, open [http://localhost:3003](http://localhost:3003) in your browser.
```

File: portafly/README.md (hunk 1 above)
Comment (cntlsn) [reply portafly/README.md:6]: what would be our suggested way?

File: portafly/README.md (hunk 2 above)
Comment (cntlsn) [reply portafly/README.md:8]: so we suggest running Porta locally first in order to be able to generate the access token, correct?

File: portafly/README.md (hunk 3 above)
Comment (cntlsn) [reply portafly/README.md:9]: you wouldn't be able to access portafly without an access token right now, so I think it is a necessary step, although temporary. we will remove it when things change.

File: portafly/README.md (hunk 4 above)
Comment (cntlsn) [reply portafly/README.md:18]: I agree with adding the disclaimer.
What other modes besides development mode are available?

File: portafly/README.md (hunk 5 above)
Comment (cntlsn) [reply portafly/README.md:24]: I know, but that doesn't look like a step by step guide on how to run portafly. It rather looks like a list of commands you can eventually run. We can link to that section for further details.

File: portafly/README.md (hunk 7 above)
Comment (cntlsn) [reply portafly/README.md:19]: agreed, thank you for linking that!

File: portafly/README.md (hunk 2 above)
Comment (Martouta) [reply portafly/README.md:8]: yeah I prefer we use it locally only

File: portafly/README.md [hunk 9]
```diff
@@ -1,6 +1,31 @@
...
+  * Select all scopes
+  * Set permissions to Read & Write
+  * Create the access token
+  * Copy the access token
+4. Allow CORS requests in your browser:
+  * Install the CORS browser extension ([Google Chrome extension](https://chrome.google.com/webstore/detail/allow-cors-access-control/lhobafahddgcelffkeicbaginigeejlf))
+  * Click on the extension icon in the browser  toolbar and activate it by clicking on its logo
+5. Run PortaFly: 
+  * Open a terminal session
+  * Navigate to `/portafly`
```
Comment (josemigallas) [inline portafly/README.md:23]: Use commands directly, less explaining more doing IMO:
`$ cd portafly`

**Pull request closed without merging**